src/Nethermind/Nethermind.Core/Extensions/EnumerableExtensions.cs

```diff
@@ -12,6 +12,22 @@ public static class EnumerableExtensions
         public static ISet<T> AsSet<T>(this IEnumerable<T> enumerable) =>
             enumerable as ISet<T> ?? enumerable.ToHashSet();
 
+        public static IEnumerable<(T item, int index)> WithIndex<T>(this IEnumerable<T> self) =>
+            self.Select((item, index) => (item, index));
+
+        public static IEnumerable<TSource> Skip<TSource>(this IEnumerable<TSource> source, ulong count)
+        {
+            IEnumerable<TSource> res = source;
+
+            while (count > int.MaxValue)
+            {
+                res = res.Skip(int.MaxValue);
+                count -= int.MaxValue;
+            }
+            res = source.Skip((int)count);
+            return res;
+        }
+
```

Comment:
Is this needed? we won't have more than int.MaxValue elements ever.

Reply:
The Shutter spec uses a ulong for the amount to skip, why won't we have more than int.MaxValue elements?

Reply:
You really expect in-memory lists over 2 bln elements?